### PR TITLE
Required libraries can be installed silently #492

### LINKED ISSUE #492: Required libraries can be installed silently
**Description**

Sequence of first time use: 
1. Shortcut
2. OK in main dialog
3. Yes for "selected test framework is not installed..." dialog
4. OK for "Download Library from Maven Repository..." dialog
5. Yes for "mock framework is not installed..." dialog
6. OK for another "Download Library from Maven Repository" dialog
7. Yes for Mockito static methods
Actually we can skip steps 3-7 with providing own JavaProjectModelModifier and getting rid of own extra dialogs

**Expected behavior**

After red warnings "will be installed" we install chosen libraries with no additional dialogs

**Context**

Plugin UX, first time use

### PR DESCRIPTION
**Pull request by Vassiliy-Kudryashov**
# Description

Getting rid of all needless Yes/No and OK/Cancel dialogs that dramatically slowdown first time use case

Fixes #492 

## Type of Change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

## Manual Scenario 

Prepare environment with clean config (e.g. no required libraries (JUnit/Mockito etc. in dependencies, empty or absent resources folder) and run tests generation. Single OK click does everything now.

# Checklist (remove irrelevant options):

- [x] The change followed the style guidelines of the UTBot project
- [x] Self-review of the code is passed
- [x] No new warnings

### CODE REVIEW

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+class UtProjectModelModifier(val project: Project) : IdeaProjectModelModifier(project) {
+    override fun addExternalLibraryDependency(
+        modules: Collection<Module?>,
+        descriptor: ExternalLibraryDescriptor,
+        scope: DependencyScope
+    ): Promise<Void> {
+        val defaultRoots = descriptor.libraryClassesRoots
+        val classesRoots: List<String?>
+        val firstModule = ContainerUtil.getFirstItem(modules)
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:37]: I do not understand the meaning of this object. Is it first not null one? We we should do something basing on it?

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import org.jetbrains.jps.model.library.JpsMavenRepositoryLibraryDescriptor
+
+class UtProjectModelModifier(val project: Project) : IdeaProjectModelModifier(project) {
+    override fun addExternalLibraryDependency(
+        modules: Collection<Module?>,
+        descriptor: ExternalLibraryDescriptor,
+        scope: DependencyScope
+    ): Promise<Void> {
+        val defaultRoots = descriptor.libraryClassesRoots
+        val classesRoots: List<String?>
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:36]: Why don't we join the declaration and assignment?

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        val firstModule = ContainerUtil.getFirstItem(modules)
+        classesRoots = if (defaultRoots.isNotEmpty()) {
+            LocateLibraryDialog(firstModule!!, defaultRoots, descriptor.presentableName).showAndGetResult()
+        } else {
+            val version = descriptor.preferredVersion
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:41]: Do you really need this variable?

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            val mavenCoordinates = descriptor.libraryGroupId + ":" +
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:42]: I suggest to create and extension method for ExternalLibraryDescriptor calculating these coordinates,

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                    descriptor.libraryArtifactId + ":" +
+                    (version ?: RepositoryLibraryDescription.ReleaseVersionId)
+
+            val libraryProperties = RepositoryLibraryProperties(JpsMavenRepositoryLibraryDescriptor(mavenCoordinates))
+            val roots = JarRepositoryManager.loadDependenciesModal(
+                project,
+                libraryProperties,
+                false,
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:50]: Our code style suggests to use named arguments for nulls, booleans, etc.

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                false,
+                null,
+                null
+            )
+            if (roots.isEmpty()) {
+                @Suppress("SpellCheckingInspection")
+                Messages.showErrorDialog(
+                    project, JavaUiBundle.message("dialog.mesage.0.was.not.loaded", descriptor.presentableName),
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:58]: Let's place one argument in one line here

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+                    JavaUiBundle.message("dialog.title.failed.to.download.library")
+                )
+                return rejectedPromise()
+            }
+            roots.stream()
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:63]: Code fragment in lines 63-70 requires comments and possibly should be moved to a separate function.
I understand nothing here. I expect that an arbitrary new intern will have no chance too.

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+                .filter { root: OrderRoot -> root.type === OrderRootType.CLASSES }
+                .map { root: OrderRoot ->
+                    PathUtil.getLocalPath(
+                        root.file
+                    )
+                }
+                .collect(Collectors.toList())
+        }
+        if (classesRoots.isNotEmpty()) {
+            val libraryName = if (classesRoots.size > 1) descriptor.presentableName else null
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:73]: Why null in else block? What is the expected behaviour for libraryName == null?

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+                    model.commit()
+                    for (module in modules) {
+                        ModuleRootModificationUtil.addDependency(module!!, library, scope, false)
+                    }
+                }
+            }
+        }
+        return resolvedPromise()
+    }
+}
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:96]: As far as I remember, you had some logic to place a dependency in the correct place (for example, after conflicting one) in build script file. Shouldn't we place that code into this function too?

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+            if (modules.size == 1) {
+                ModuleRootModificationUtil.addModuleLibrary(firstModule!!, libraryName, urls, emptyList(), scope)
+            } else {
+                WriteAction.run<RuntimeException> {
+                    val library =
+                        LibraryUtil.createLibrary(
+                            LibraryTablesRegistrar.getInstance().getLibraryTable(project), descriptor.presentableName
+                        )
+                    val model = library.modifiableModel
+                    for (url in urls) {
```
Comment (EgorkaKulikov) [inline utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:84]: urls.filterNotNull().forEach { ... }

**Review by EgorkaKulikov — APPROVED** [review]

**Comment by EgorkaKulikov** [comment]
@alisevych I suggest to test this functionality carefully in different projects (Maven, Gradle, Idea) of different versions. Potential breaking changes.

File: utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt (hunk 9 above)
Comment (Vassiliy-Kudryashov) [reply utbot-intellij/src/main/kotlin/org/utbot/intellij/plugin/util/UtProjectModelModifier.kt:85]: New UtProjectModelModifier works at stage GenerateTestDialogWindow.addDependency() so the next step org.utbot.intellij.plugin.ui.GenerateTestsDialogWindow#placeEntryToCorrectPlace works too inside the same stage

**Pull request merged**